service/node.go

```diff
@@ -2671,12 +2672,29 @@ func (s *service) NodeExpandVolume(
 		deviceNames, _ := gofsutil.GetSysBlockDevicesForVolumeWWN(context.Background(), volumeWWN)
 		if len(deviceNames) > 0 {
 			for _, deviceName := range deviceNames {
-				devicePath := sysBlock + "/" + deviceName
-				log.Infof("Rescanning unmounted (raw block) device %s to expand size", deviceName)
-				err = gofsutil.DeviceRescan(context.Background(), devicePath)
-				if err != nil {
-					log.Errorf("Failed to rescan device (%s) with error (%s)", devicePath, err.Error())
-					return nil, status.Error(codes.Internal, err.Error())
+				if strings.HasPrefix(deviceName, "nvme") {
+					nvmeControllerDevice, err := gofsutil.GetNVMeController(deviceName)
+					if err != nil {
+						log.Errorf("Failed to rescan device (%s) with error (%s)", deviceName, err.Error())
+						return nil, status.Error(codes.Internal, err.Error())
+					}
```

Comment:
Why would the GetNVMeController() call fail? Is bailing out the proper handling? Could anything else be done? The error message is inaccurate and can cause problems during troubleshooting. Is it better to indicate that there was a failure to get the device controller?